pandas/core/arrays/datetimelike.py

```diff
@@ -1679,3 +1679,27 @@ def _ensure_datetimelike_to_i8(other, to_utc=False):
             # period array cannot be coerced to int
             other = Index(other)
     return other.asi8
+
+
+def is_valid_na(obj, dtype):
+    """
+    isna check that excludes incompatible dtypes
+
+    Parameters
+    ----------
+    obj : object
+    dtype : np.datetime64, np.timedelta64, DatetimeTZDtype, or PeriodDtype
+
+    Returns
+    -------
+    bool
+    """
+    if not isna(obj):
+        return False
+    if dtype.kind == "M":
+        return not isinstance(obj, np.timedelta64)
```

Comment:
and why isn't NaT valid here?

Reply:
NaT _is_ valid here.  In fact, NaT should return True for all three cases

Reply:
right, so you can eliinate from the condition where you are calling this